ArduCopter/AP_Arming.cpp

```diff
@@ -643,6 +671,15 @@ bool AP_Arming_Copter::arm_checks(AP_Arming::Method method)
     return AP_Arming::arm_checks(method);
 }
 
+// mandatory checks that will be run if ARMING_CHECK is zero or arming forced
+bool AP_Arming_Copter::mandatory_checks(bool display_failure)
```

Comment:
This confuses me because we already have some mandatory checks in `bool AP_Arming_Copter::run_pre_arm_checks(bool display_failure)`  (https://github.com/ardupilot/ardupilot/blob/master/ArduCopter/AP_Arming.cpp#L38)

If force-arming bypassed those then I think we're agreed there's a bug as forced-arming should be the same as setting the arming-check bitmask to zero.  If we have such a bug then the checks I've referenced aren't being run, and that's not good.

Reply:
@peterbarker, yes and I struggled with all this going back and forth on where to put the "mandatory_gps_checks" and the interlock checks.  The problem is that beyond providing the failure message to the user we also need to set the "AP_Notify::flags.pre_arm_gps_check" flag.  If ARMING_CHECK = 0 or using the "force" armed then the flag needs to be set based on the shorter set of checks, otherwise the flag needs to be set based on the full set.  There are a couple of ways to accomplish this but I thought this method was as good as any.